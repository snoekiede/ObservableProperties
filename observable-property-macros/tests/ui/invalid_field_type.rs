use observable_property::Observable;

// This should fail to compile because tuple structs aren't supported by #[derive(Observable)]
#[derive(Observable)]
struct BadStruct(String, i32);

fn main() {}
