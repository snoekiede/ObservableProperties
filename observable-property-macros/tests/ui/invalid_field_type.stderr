error: Observable can only be derived for structs with named fields
 --> tests/ui/invalid_field_type.rs:5:1
  |
5 | struct BadStruct(String, i32);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
